src/data/matrix/basic.lean

```diff
@@ -1405,15 +1413,25 @@ def transpose_ring_equiv [add_comm_monoid α] [comm_semigroup α] [fintype m] :
     (mul_opposite.op_mul _ _),
   ..transpose_add_equiv.trans mul_opposite.op_add_equiv }
 
-variables {m α}
-
 @[simp] lemma transpose_pow [comm_semiring α] [fintype m] [decidable_eq m] (M : matrix m m α)
   (k : ℕ) : (M ^ k)ᵀ = Mᵀ ^ k :=
-mul_opposite.op_injective $ map_pow (transpose_ring_equiv m α) M k
+mul_opposite.op_injective $ map_pow (transpose_ring_equiv : matrix m m α ≃+* _) M k
```

Comment:
This type of change is usually a good indication that the arguments should have been explicit

Reply:
It seems much of a muchness to me. We already did this, in this file, for `transpose_add_equiv`. I'm happy to change it so that they're both explicit! But they should be consistent, and if this isn't OK for `ring`, it shouldn't be for `add`.

Reply:
I just committed an all-explicits commit. That might have broken something but if it did I can roll it back. What do you think?

Reply:
The "all-explicits" version looks like an improvement to me; you'll need to fix a few things downstream though.

Reply:
Ah, I'd rolled this back as it had broken things and I'd had no comment on it. I'll check it builds without it, and then apply it and fix the downstream errors.

Reply:
(What's the best way of finding the downstream errors? Just search for the use of these equivs?)

Reply:
Letting CI find them. CI already found some downstream errors from the commit you pushed before.

Reply:
I actually recommend you force push away your reversion commit, as that will save you half an hour of rebuilding to get to the state you already had in the previous commit.

Reply:
Right, so just iterate using CI. Can do.

Reply:
You can of course build locally too

Reply:
My computer doesn't seem good enough.

Reply:
`lean --make src/the_file/that_I_think_might_break_next` often doesn't take too long; mathlib is likely wider than it is deep, so build times are a lot faster if you only care about one file.

Reply:
Right, let's see how that is.